daffodil-lib/src/main/scala/org/apache/daffodil/api/XMLOutputStyle.scala

```diff
@@ -0,0 +1,26 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.daffodil.api
+
+import org.apache.daffodil.util.Enum
+
+object XMLOutputStyle extends Enum {
```

Comment:
For consistency, I think we want specific japi and sapi variants if this enum. This ensures that the documentation actually ends up in the Java/Scaladocs (no docs are generated from daffodil-lib) and ensures that all the "public" API is all in the .japi or .sapi namespace.

For example, we have the ValidationMode enum which is duplicated in the japi/sapi, and then there is a ValidationConversions object to convert to the japi/sapi enums to the api enum that is actually used by the underlying XMLTextInfosetOutputter. It's not the prettiest thing in the world, but it does ensure that all the public avialble objects are in the .japi and .sapi namespaces and end up in the java/scala doc.

I also wonder if this enum wants to be in the .infoset namespace instead of the .api namespace, since it really is only ever going to be used for infoset clasess?